ocf_data_sampler/config/model.py

```diff
@@ -0,0 +1,236 @@
+
+
+class General(Base):
+    """General pydantic model"""
+
+    name: str = Field("example", description="The name of this configuration file.")
+    description: str = Field(
+        "example configuration", description="Description of this configuration file"
+    )
+
+
+class DataSourceMixin(Base):
+    """Mixin class, to add forecast and history minutes"""
+
+    forecast_minutes: int = Field(
+        None,
+        ge=0,
+        description="how many minutes to forecast in the future. "
+        "If set to None, the value is defaulted to InputData.default_forecast_minutes",
+    )
+    history_minutes: int = Field(
+        None,
+        ge=0,
+        description="how many historic minutes to use. "
+        "If set to None, the value is defaulted to InputData.default_history_minutes",
+    )
+
+
+# noinspection PyMethodParameters
+class DropoutMixin(Base):
+    """Mixin class, to add dropout minutes"""
+
+    dropout_timedeltas_minutes: Optional[List[int]] = Field(
+        default=None,
+        description="List of possible minutes before t0 where data availability may start. Must be "
+        "negative or zero.",
+    )
+
+    dropout_fraction: float = Field(0, description="Chance of dropout being applied to each sample")
+
+    @field_validator("dropout_timedeltas_minutes")
+    def dropout_timedeltas_minutes_negative(cls, v: List[int]) -> List[int]:
+        """Validate 'dropout_timedeltas_minutes'"""
+        if v is not None:
+            for m in v:
+                assert m <= 0
+        return v
+
+    @field_validator("dropout_fraction")
+    def dropout_fraction_valid(cls, v: float) -> float:
+        """Validate 'dropout_fraction'"""
+        assert 0 <= v <= 1
+        return v
+
+
+# noinspection PyMethodParameters
+class TimeResolutionMixin(Base):
+    """Time resolution mix in"""
+
+    time_resolution_minutes: int = Field(
+        ...,
+        description="The temporal resolution of the data in minutes",
+    )
+
+
+class Satellite(DataSourceMixin, TimeResolutionMixin, DropoutMixin):
+    """Satellite configuration model"""
+
+    # Todo: remove 'satellite' from names
+    satellite_zarr_path: Union[str, tuple[str], list[str]] = Field(
+        ...,
+        description="The path or list of paths which hold the satellite zarr",
+    )
+    satellite_channels: tuple = Field(
+        ..., description="the satellite channels that are used"
+    )
+    satellite_image_size_pixels_height: int = Field(
+        ...,
+        description="The number of pixels of the height of the region of interest"
+        " for non-HRV satellite channels.",
+    )
+
+    satellite_image_size_pixels_width: int = Field(
+        ...,
+        description="The number of pixels of the width of the region "
+        "of interest for non-HRV satellite channels.",
+    )
+
+    live_delay_minutes: int = Field(
+        30, description="The expected delay in minutes of the satellite data"
```

Comment:
I think this should not have a default value and should be required

Reply:
I think `live` was all being stripped out

Reply:
It is but not until we switch to `interval_[start/end]`

Reply:
this is just how delay/dropout is organised in sat currently, generally yes I think we are moving towards a) calling delay, especially for nwp, what it is (#18) and/or b) removing the need for that altogether with `interval_start/interval_end` (#22). As far as I understand it doesn't actually have anything to do with live as the environment, its live in a sense that this is the delay we expect to have in 'live' circumstances (do correct me if I'm wrong!) 

This is still needed so I think we're keeping it in until we resolve either of the issues above; we can rename it in the meantime but with it being scheduled to be chucked maybe not worth it